### PR TITLE
(SIMP-1695) Update global catalysts

### PR DESCRIPTION
**Pull request by nick-markowski**
- Strong typed, as well.

SIMP-2176 #close
SIMP-2292 #close

### CODE REVIEW

**Comment by nick-markowski** [comment]
This won't pass travis untill stunnel is push through.

File: .puppet-lint.rc [hunk 1]
```diff
@@ -2,5 +2,6 @@
 --relative
 --no-class_inherits_from_params_class-check
 --no-80chars-check
```
Comment (lnemsick-simp) [inline .puppet-lint.rc]: According to Nick Miller, if we have no-140chars-check, we don't need to specify no-80chars-check

File: README.md [hunk 2]
```diff
@@ -86,20 +85,20 @@ path:
 
 ``` puppet
 class site::nfs_server (
-  $usekrb5 = hiera('nfs::simp_krb5',false),
-  $client_nets = defined('$::client_nets') ? { true => $::client_nets, default => hiera('client_nets') }
+  $kerberos = simplib::lookup('simp_options::kerberos', { 'default_value' => false, 'value_type' => Boolean }),
+  $trusted_nets = defined('$::trusted_nets') ? { true => $::trusted_nets, default => hiera('trusted_nets') }
```
Comment (lnemsick-simp) [inline README.md]: shouldn't this be simp_options::trusted_nets?

File: README.md [hunk 3]
```diff
@@ -86,20 +85,20 @@ path:
...
   ){
   include '::nfs'
 
-  if $usekrb5 {
+  if $kerberos {
```
Comment (lnemsick-simp) [inline README.md]: if block and the $security = $kerberos ? ... statement look to be the same

File: spec/acceptance/suites/stunnel/03_stunnel_test_spec.rb [hunk 4]
```diff
@@ -38,7 +38,7 @@
 ---
 # Need to test that our iptables works properly with this stuff
 
-use_iptables : true
+firewall : true
```
Comment (lnemsick-simp) [inline spec/acceptance/suites/stunnel/03_stunnel_test_spec.rb]: In the acceptance tests, why isn't the hieradata using simp_options, e.g.
 simp_options::trusted_nets
 simp_options::firewall
 simp_options::auditd

File: templates/nfs_sysconfig_server.erb [hunk 5]
```diff
@@ -4,32 +4,32 @@ RQUOTAD="<%= scope.lookupvar('::nfs::rquotad') %>"
 RQUOTAD=no
 <% end -%>
 RQUOTAD_PORT=<%= scope.lookupvar('::nfs::rquotad_port') %>
-<% if @rpcrquotadopts && has_variable?('rpcrquotadopts') && !@rpcrquotadopts.empty? -%>
+<% if @rpcrquotadopts -%>
 RPCRQUOTADOPTS="<%= @rpcrquotadopts %>"
 <% end -%>
-<% if @lockd_arg && has_variable?('lockd_arg') && !@lockd_arg.empty? -%>
+<% if @lockd_arg -%>
 LOCKD_ARG="<= lockd_arg %>"
```
Comment (nick-markowski) [inline templates/nfs_sysconfig_server.erb]: Looks like it's missing the interpolate arg % and proper scoping, '@'

File: manifests/server/export.pp [hunk 6]
```diff
@@ -12,116 +14,89 @@
...
+  Boolean                                         $no_wdelay      = false,
+  Boolean                                         $nohide         = false,
+  Boolean                                         $crossmnt       = false,
+  Boolean                                         $subtree_check  = false,
+  Boolean                                         $insecure_locks = false,
+  Optional[Variant[Boolean,Stdlib::Absolutepath]] $mountpoint     = undef,
+  Optional[String]                                $fsid           = undef,
+  Boolean                                         $nordirplus     = false,
+  Optional[String]                                $refer          = undef,
+  Tuple[Enum['sys','krb5','krb5i','krb5p'],1,2]   $sec            = ['sys'],
```
Comment (lnemsick-simp) [inline manifests/server/export.pp]: Is this tuple trying to restrict to 1 or 2 items in the list?

File: spec/acceptance/nodesets/default.yml [hunk 7]
```diff
@@ -60,5 +60,5 @@ HOSTS:
 
 CONFIG:
   log_level: verbose
-  type: foss
+  type: aio
```
Comment (nick-markowski) [inline spec/acceptance/nodesets/default.yml]: I should probably add this and ensuring the gemfile contains Trevor's beaker patch to the 1695 instructions

File: README.md (hunk 2 above)
Comment (nick-markowski) [inline README.md]: yes, but I basically sed-ed the readme. I'm waiting to do full reconstructs in SIMP-2213.

File: manifests/server/export.pp (hunk 6 above)
Comment (nick-markowski) [inline manifests/server/export.pp]: The idea was min and max, so both. However, this should actually check to make sure only one type of krb* is passed.  Sys and one type of krb5 is fine, but not >1 krb5.

**Comment by nick-markowski** [comment]
Patch incoming soon...

**Pull request merged**